marketplaceItems/avatarCustomization101/appResources/appData/resources/modules/mirrorClient.js

```diff
@@ -35,18 +36,24 @@
     // LOCAL FUNCTIONS
     
     // When x or y dimensions of the mirror change - reset the resolution of the 
-    // spectator camera and edit the mirror overlay to adjust for the new dimensions
+    // spectator camera and edit the mirror local entity to adjust for the new dimensions
     function updateMirrorDimensions(forceUpdate) {
-        if (mirrorOverlayRunning) {
-            var newDimensions = Entities.getEntityProperties(_this.entityID, 'dimensions').dimensions;
 
-            if (forceUpdate === true || (newDimensions.x !== lastDimensions.x || newDimensions.y !== lastDimensions.y)) {
+        if (!mirrorLocalEntityID) {
+            return;
+        }
+        
+        if (mirrorLocalEntityRunning) {
+          
+            var newDimensions = Entities.getEntityProperties(_this.entityID, 'dimensions').dimensions;
+            
+            if (forceUpdate === true || (newDimensions.x != lastDimensions.x || newDimensions.y != lastDimensions.y)) {
                 var mirrorResolution = _this.calculateMirrorResolution(newDimensions);
                 spectatorCameraConfig.resetSizeSpectatorCamera(mirrorResolution.x, mirrorResolution.y);
-                Overlays.editOverlay(mirrorOverlayID, {
+                Entities.editEntity(mirrorLocalEntityID, {
                     dimensions: {
-                        x: -(newDimensions.y > newDimensions.x ? newDimensions.y : newDimensions.x),
-                        y: -(newDimensions.y > newDimensions.x ? newDimensions.y : newDimensions.x),
+                        x: (Math.max(newDimensions.x, newDimensions.y)),
```

Comment:
Does this still act like a mirror? The -1 multiplier used to be necessary for that.

Reply:
Ah, is that what your `localRotation: Quat.fromPitchYawRollDegrees(0, 0, 180),` is doing below?

Reply:
yes, still behaves like a mirror. quat thing was because it was rendering upside down for some reason